### PR TITLE
Bld 33537

### PR DESCRIPTION
**Pull request by ozbenoved**

### CODE REVIEW

File: botdemo/app/build.gradle [hunk 1]
```diff
@@ -7,17 +7,15 @@ buildscript {
     }
 
     dependencies {
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.31"
-        classpath "org.jetbrains.kotlin:kotlin-android-extensions:1.3.31"
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.60"
```
Comment (tehilaR2) [inline botdemo/app/build.gradle]: can be 1.3.72

File: botdemo/app/build.gradle [hunk 2]
```diff
@@ -31,18 +29,26 @@ android {
...
     }
 
+    testBuildType "sdktesting"
+
     buildTypes {
         release {
             minifyEnabled false
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
+        sdktesting {
```
Comment (tehilaR2) [inline botdemo/app/build.gradle]: why do u need this - thats for the unit testing

File: botdemo/app/src/main/java/nanorep/com/botdemo/MainActivity.kt [hunk 3]
```diff
@@ -204,15 +221,15 @@ class MainActivity : AppCompatActivity(), ChatHandler {
...
-        if (readoutEnabled) {
-            settings.enableReadOut()
-        }
-
         return ChatController.Builder(this).apply {
             conversationSettings(settings)
             chatEventListener(this@MainActivity)
-            chatHandoverHandler(myHandoverHandler);
-            entitiesProvider(entitiesProvider).apply {  }
+            chatHandoverHandler(handoverHandler);
```
Comment (tehilaR2) [inline botdemo/app/src/main/java/nanorep/com/botdemo/MainActivity.kt]: ;?

File: botdemo/app/src/main/java/nanorep/com/botdemo/MainActivity.kt [hunk 4]
```diff
@@ -204,15 +221,15 @@ class MainActivity : AppCompatActivity(), ChatHandler {
...
+            entitiesProvider(entitiesProvider)
+            if (readoutEnabled) {
```
Comment (tehilaR2) [inline botdemo/app/src/main/java/nanorep/com/botdemo/MainActivity.kt:229]: where is it coming from ? - if this is a configuration issue - we're checking this on the handlers - don't we?

**Review by tehilaR2 — CHANGES_REQUESTED** [review]

File: botdemo/app/src/main/java/nanorep/com/botdemo/MainActivity.kt (hunk 4 above)
Comment (ozbenoved) [reply botdemo/app/src/main/java/nanorep/com/botdemo/MainActivity.kt:229]: Its an option at the ui (checkbox)

File: botdemo/app/src/main/java/nanorep/com/botdemo/MainActivity.kt (hunk 3 above)
Comment (ozbenoved) [reply botdemo/app/src/main/java/nanorep/com/botdemo/MainActivity.kt]: removed

File: botdemo/app/build.gradle (hunk 2 above)
Comment (ozbenoved) [reply botdemo/app/build.gradle]: removed

File: botdemo/app/build.gradle (hunk 1 above)
Comment (ozbenoved) [reply botdemo/app/build.gradle]: updated

**Review by tehilaR2 — APPROVED** [review]

**Pull request merged**